075_Inside_a_shard/30_Dynamic_indices.asciidoc

```diff
@@ -1,86 +1,64 @@
 
 [[img-index-segments]]
-.A Lucene index with a commit point and three segments
+.一个 Lucene 索引包含一个提交点和三个段
 image::images/elas_1101.png["A Lucene index with a commit point and three segments"]
 
-.Index Versus Shard
+.索引与分片
 ***************************************
 
-To add to the confusion, a _Lucene index_ is what we call a _shard_ in
-Elasticsearch, while an _index_ in Elasticsearch((("indices", "in Elasticsearch")))((("shards", "indices versus"))) is a collection of shards.
-When Elasticsearch searches an index, it sends the query out to a copy of
-every shard (Lucene index) that belongs to the index, and then reduces the
-per-shards results to a global result set, as described in
-<<distributed-search>>.
+被混淆的概念是，一个 _Lucene 索引_ 我们在Elasticsearch称作 _分片_ 。
+一个 Elasticsearch _索引_ ((("indices", "in Elasticsearch")))((("shards", "indices versus"))) 是分片的集合。
+当 Elasticsearch 在索引中搜索的时候， 他发送查询到每一个属于索引的分片(Lucene 索引)，然后像<<distributed-search>>
+提到的那样，合并没个分片的结果到一个全局的结果集。
 
 ***************************************
 
+每单个段搜索像下面提到的工作：
 
-A per-segment search works as follows:
-
-1. New documents are collected in an in-memory indexing buffer.
-   See <<img-memory-buffer>>.
-2. Every so often, the buffer is _commited_:
+1. 新文档被收集到内存索引缓存， 见<<img-memory-buffer>>。
+2. 经常性的, 缓存被 _提交_:
 
-** A new segment--a supplementary inverted index--is written to disk.
-** A new _commit point_ is written to disk, which includes the name of the new
-   segment.
-** The disk is _fsync'ed_&#x2014;all writes waiting in the filesystem cache are
-   flushed to disk, to ensure that they have been physically written.
-
-3. The new segment is opened, making the documents it contains visible to search.
-4. The in-memory buffer is cleared, and is ready to accept new documents.
+** 一个新的段--补充倒排索引--被写入磁盘。
+** 一个新的包含新段名字的 _提交点_ 被写入磁盘。
+** 一个磁盘是 _fsync'ed_&#x2014; 所有写入等待文件系统缓存被刷入磁盘，来保证它们被物理写入。
+3. 一个新的段开启，让它包含的文档可见以被搜索。
+4. 内存缓存被清空，等待接收新的文档。
 
 [[img-memory-buffer]]
-.A Lucene index with new documents in the in-memory buffer, ready to commit
+.一个 Lucene 有内存缓存新文档的索引，准备提交:
 image::images/elas_1102.png["A Lucene index with new documents in the in-memory buffer, ready to commit"]
 
 [[img-post-commit]]
-.After a commit, a new segment is added to the commit point and the buffer is cleared
+.在一次提交后，一个新的段被添加到提交点而且缓存被清空。
 image::images/elas_1103.png["After a commit, a new segment is added to the index and the buffer is cleared"]
 
-When a query is issued, all known segments are queried in turn. Term
-statistics are aggregated across all segments to ensure that the relevance of
-each term and each document is calculated accurately. In this way, new
-documents can be added to the index relatively cheaply.
+当一个查询被发布，所有的已知的段反过来被查询。词统计汇通过所有段去聚合，以保证每个词和每个文档的关联都被准确计算。
```

Comment:
非常抱歉，我看到
When a query is issued, all known segments are queried in turn
+当一个查询被发布，所有的已知的段反过来被查询
这里in turn应该是“按次序“。


kevin@game4u.cn
 
From: 曹春晖
Date: 2016-08-08 16:27
To: elasticsearch-cn/elasticsearch-definitive-guide
Subject: Re: [elasticsearch-cn/elasticsearch-definitive-guide] chapter11_part3:/075_Inside_a_shard/30_Dynamic_indices.asciidoc (#167)
In 075_Inside_a_shard/30_Dynamic_indices.asciidoc:
>  image::images/elas_1103.png["After a commit, a new segment is added to the index and the buffer is cleared"]
>  
> -When a query is issued, all known segments are queried in turn. Term
> -statistics are aggregated across all segments to ensure that the relevance of
> -each term and each document is calculated accurately. In this way, new
> -documents can be added to the index relatively cheaply.
> +当一个查询被发布，所有的已知的段反过来被查询。词统计汇通过所有段去聚合，以保证每个词和每个文档的关联都被准确计算。

发布=>触发
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.